File: gaia-sdk-javascript/src/graphql/request/input/PerceiveButtonImpulse.ts


import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * Input for button perception impulse
 */
export class PerceiveButtonImpulse {
public _typeName = "PerceiveButtonImpulse";
    private name:String;
    private value:String;

    constructor (name:String, value:String) {
        this.name = name;
        this.value = value;
    }
}

File: gaia-sdk-python/gaia_sdk/graphql/request/input/PerceiveButtonImpulse.py


from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class PerceiveButtonImpulse():
    """
    Input for button perception impulse
    """
    name: str
    value: str

    def __init__(self, name: str, value: str):
        self.name = name
        self.value = value

    def __eq__(self, other):
        if type(other) is type(self):
            return self.name == other.name and self.value == other.value
        return False

    def __repr__(self):
        return {'name': self.name, 'value': self.value}
